File: backend/src/main/java/hu/bsstudio/raktr/dto/ticket/TicketCreateDto.java
package hu.bsstudio.raktr.dto.ticket;

import hu.bsstudio.raktr.dal.value.ProblemSeverity;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TicketCreateDto {

    @NotBlank
    private String description;

    @NotNull
    private ProblemSeverity severity;

    @Min(1)
    @NotNull
    private Long scannableId;

}

File: frontend/src/app/model/ticket/ticketCreateDto.ts
import {TicketSeverity} from './ticketSeverity';

export class TicketCreateDto {
  description: string;
  severity: TicketSeverity;
  scannableId: number;

  constructor(
    description: string,
    severity: TicketSeverity,
    scannableId: number
  ) {
    this.description = description;
    this.severity = severity;
    this.scannableId = scannableId;
  }

  toJson(): Record<string, unknown> {
    return {
      description: this.description,
      severity: this.severity,
      scannableId: this.scannableId
    };
  }
}
